### PR TITLE
config: add config variable for reply to mail attribute

### LINKED ISSUE #19: add no-reply sender address configuration
As noticed on user support, the users have a habit of replying on the system generated e-mails, therefore we need a configuration a common no-reply email address for the notifications sent by the module

### PR DESCRIPTION
**Pull request by rekt-hard**
:heart: Thank you for your contribution!

### Description

closes #19 
depends on https://github.com/inveniosoftware/invenio-mail/pull/59

Set `reply_to` attribute in the e-mail backend.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [ ] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [ ] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [ ] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_notifications/config.py [hunk 1]
```diff
@@ -57,3 +57,6 @@
 This should be set higher up in the module hierarchy (e.g. invenio-app-rdm), as
 this module does not have knowledge of the settings view.
 """
+
+NOTIFICATIONS_MAIL_REPLY_TO = None
+"""Reply to mail address for notifications."""
```
Comment (zzacharo) [inline invenio_notifications/config.py:62]: maybe add a comment that if None what would be the mail sender? Also, after testing, maybe add here the comment about `MAIL_DEFAULT_SENDER`?

Comment (rekt-hard) [reply invenio_notifications/config.py:62]: The mail sender is always `MAIL_DEFAULT_SENDER`. This is only for the `reply to` attribute of the email. I will put some explanations in the comment.

File: invenio_notifications/config.py [hunk 2]
```diff
@@ -57,3 +57,11 @@
 This should be set higher up in the module hierarchy (e.g. invenio-app-rdm), as
 this module does not have knowledge of the settings view.
 """
+
+NOTIFICATIONS_MAIL_REPLY_TO = None
```
Comment (kpsherva) [inline invenio_notifications/config.py:61]: I think it could be global reply_to attribute, not only notifications

Comment (zzacharo) [inline invenio_notifications/config.py:61]:
```suggestion
MAIL_DEFAULT_REPLY_TO = None
```

File: invenio_notifications/backends/email.py [hunk 3]
```diff
@@ -33,6 +33,7 @@ def send(self, notification, recipient):
                     recipient.data.get("email") or recipient.data.get("email_hidden")
                 ],
                 "sender": current_app.config["MAIL_DEFAULT_SENDER"],
+                "reply_to": current_app.config["NOTIFICATIONS_MAIL_REPLY_TO"],
```
Comment (zzacharo) [inline invenio_notifications/backends/email.py:36]: see rename below to be in align with the `MAIL_DEFAULT_SENDER`

**Review by zzacharo — APPROVED** [review]

**Comment by rekt-hard** [comment]
moved the config value to `invenio-mail`

**Pull request merged**